mobile/geth.go

```diff
@@ -185,7 +217,56 @@ func NewNode(datadir string, config *NodeConfig) (stack *Node, _ error) {
 			return nil, fmt.Errorf("whisper init: %v", err)
 		}
 	}
-	return &Node{rawStack}, nil
+	if config.PssEnabled && ks != nil {
+		log.Debug("pss enabled")
+		bzzSvc := func(ctx *node.ServiceContext) (node.Service, error) {
+			log.Debug("keystore", "ks", ks)
+			var a accounts.Account
+			var err error
+			if common.IsHexAddress(config.PssAccount) {
+				a = ks.GetAccounts().accounts[0]
+			} else if ix, ixerr := strconv.Atoi(config.PssAccount); ixerr == nil && ix > 0 {
+				if accounts := ks.GetAccounts().accounts; len(accounts) > ix {
+					a = accounts[ix]
+				} else {
+					err = fmt.Errorf("index %d higher than number of accounts %d", ix, len(accounts))
+				}
+			} else {
+				return nil, fmt.Errorf("Can't find swarm account key %s", config.PssAccount)
+			}
+			if err != nil {
+				return nil, fmt.Errorf("Can't find swarm account key: %v - Is the provided bzzaccount(%s) from the right datadir/Path?", err, config.PssAccount)
+			}
+			keyjson, err := ioutil.ReadFile(a.URL.Path)
+			if err != nil {
+				return nil, fmt.Errorf("Can't load swarm account key: %v", err)
+			}
+			var bzzkey *ecdsa.PrivateKey
+			key, err := keystore.DecryptKey(keyjson, config.PssPassword)
+			if err == nil {
+				bzzkey = key.PrivateKey
+			}
+			if bzzkey == nil {
+				return nil, fmt.Errorf("Can't decrypt swarm account key")
+			}
+			bzzconfig := swarmapi.NewConfig()
+			bzzconfig.SyncEnabled = false
+			bzzconfig.Path = rawStack.DataDir()
+			bzzconfig.Init(bzzkey)
+
+			svc, err := swarm.NewSwarm(bzzconfig, nil)
+			if err != nil {
+				log.Error("swarm svc", "err", err)
```

Comment:
? are you happy continuing even if swarm is not working? not sure i get it
if not return with error right away